core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -371,15 +371,30 @@ private[spark] class MesosClusterScheduler(
     envBuilder.addVariables(
       Variable.newBuilder().setName("SPARK_EXECUTOR_OPTS").setValue(executorOpts))
     val cmdOptions = generateCmdOption(desc)
+    val dockerDefined = desc.schedulerProperties.contains("spark.mesos.executor.docker.image")
     val executorUri = desc.schedulerProperties.get("spark.executor.uri")
       .orElse(desc.command.environment.get("SPARK_EXECUTOR_URI"))
     val appArguments = desc.command.arguments.mkString(" ")
-    val cmd = if (executorUri.isDefined) {
+    val (executable, jar) = if (dockerDefined) {
+      val executorOpts = desc.schedulerProperties
+        .filterKeys(_.contains("docker"))
+        .map { case (k, v) =>
+        s"-D$k=$v"
+      }.mkString(" ")
+      // Pass docker options to executors
+      envBuilder.addVariables(
+        Variable.newBuilder()
+          .setName("SPARK_EXECUTOR_OPTS")
+          .setValue(executorOpts))
+      // Application jar is automatically downloaded in the mounted sandbox by Mesos,
+      // and the path to the mounted volume is stored in $MESOS_SANDBOX env variable.
+      ("./bin/spark-submit", s"$$MESOS_SANDBOX/${desc.jarUrl.split("/").last}")
```

Comment:
I want to have a literal $MESOS_SANDBOX, with string template $ will be evaluated so I need to escape with another $